### PR TITLE
Authentication and NavLink

### PR DESCRIPTION
**Pull request by saketkum11**
- added signin api call 
- added signup api call
- login html
- signup html

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -9,34 +9,172 @@ import Home from "./Pages/Home/Home";
 import Signup from "./Pages/Auth/Signup";
 import Login from "./Pages/Auth/Login";
 import Notes from "./Pages/Notes/Notes";
+import Mockman from "mockman-js";
+import RequierAuth from "./Context/RequireAuth/RequireAuth";
+import { useAuth } from "./Context/Auth/Auth-Context";
```
Comment (shubhambishnoi) [inline src/App.js:14]: you can correct the typo in RequireAuth

**Review by shubhambishnoi — APPROVED** [review]

File: src/App.js (hunk 1 above)
Comment (saketkum11) [reply src/App.js:14]: okk sure.

Comment (saketkum11) [reply src/App.js:14]: okk sure.

File: src/Pages/Auth/Login.jsx [hunk 2]
```diff
@@ -1,9 +1,89 @@
...
 function Login() {
+  const navigate = useNavigate();
+  const location = useLocation();
+  const { auth, loginTextHandler, loginHandler } = useAuth();
+  const [userCredentials, setUserCredentials] = useState({
+    email: "",
+    password: "",
+  });
+  useEffect(() => {
+    console.log(location.state?.from?.pathname);
```
Comment (narharikale) [inline src/Pages/Auth/Login.jsx]: i think missed to remove console

File: src/Pages/Auth/Signup.jsx [hunk 3]
```diff
@@ -1,9 +1,91 @@
...
+            className="bg-purple-7 m-y-8 rounded-xs border-none outline-none text-s cursor text-color-0 pd-3 text-light"
+          >
+            Sign Up
+          </button>
+          <a
+            href="#"
+            className="text-dec rounded-s text-color-9 bg-black-2 pd-x-3 pd-y-2"
+          >
+            Already have account
+          </a>
```
Comment (narharikale) [inline src/Pages/Auth/Signup.jsx:86]: this should be Link

**Review by narharikale — APPROVED** [review]

**Review by harshitpaliwal95 — APPROVED** [review]
every thing looks good

**Review by shubhamkumar648 — APPROVED** [review]
as i saw few peers are already suggested. The remaining code and functionality is good  good to go

File: src/Pages/Auth/Login.jsx (hunk 2 above)
Comment (saketkum11) [reply src/Pages/Auth/Login.jsx]: okk I forgot to remove

File: src/Pages/Auth/Signup.jsx (hunk 3 above)
Comment (saketkum11) [reply src/Pages/Auth/Signup.jsx:86]: Yes .

**Pull request merged**